happi/cli.py

```diff
@@ -84,9 +90,50 @@ def search(
     be combined with ANDs.
     """
     logger.debug("We're in the search block")
-    # Retrieve client
-    client = ctx.obj
 
+    final_results = search_parser(
+        client=ctx.obj,
+        use_glob=use_glob,
+        search_criteria=search_criteria,
+    )
+    if not final_results:
+        return []
+
+    # Final processing for output
+    if show_json:
+        json.dump([dict(res.item) for res in final_results], indent=2,
+                  fp=sys.stdout)
+    elif names:
+        out = " ".join([res.item.name for res in final_results])
+        click.echo(out)
+    else:
+        for res in final_results:
+            res.item.show_info()
+
+    return final_results
+
+
+def search_parser(
+    client: happi.Client,
+    use_glob: bool,
+    search_criteria: List[str],
+) -> List[happi.SearchResult]:
```

Comment:
At some point maybe we apply this to the `happi load` command to keep the search interface uniform?

Reply:
Good point, I totally forgot about that

Reply:
This will need at least a thinking pause to implement- load behaves slightly differently to the other methods right now and we need to make sure we don't break anything. As such I won't finish it here.

Reply:
I was thinking about this for a future PR if anything

Reply:
For sure, it just sounded easy (<10min) to do so I poked at it and then realized it was not